src/System.Management.Automation/engine/MshCommandRuntime.cs

```diff
@@ -3055,21 +3148,38 @@ internal ActionPreference VerbosePreference
 
                 if (!_isVerbosePreferenceCached)
                 {
-                    bool defaultUsed = false;
-                    _verbosePreference = Context.GetEnumPreference(
-                        SpecialVariables.VerbosePreferenceVarPath,
-                        _verbosePreference,
-                        out defaultUsed);
+                    _verbosePreference = Context.GetEnumPreference(SpecialVariables.VerbosePreferenceVarPath, _verbosePreference, out _);
+
+                    // If the host couldn't prompt for the verbose action anyways, change it to 'Continue'.
+                    // This lets hosts still see debug output without having to implement the prompting logic.
+                    if ((CBhost.ExternalHost.UI == null) && (_verbosePreference == ActionPreference.Continue))
```

Comment:
Yes, that should be for `ActionPreference.Inquire`. I've updated that locally. I'm going to leave the logic there (but fixed to use `Inquire` because it forces `Continue` behavior if the host does not support user interaction.